detection-rules/attachment_filename_with_unicode_rtlo.yml

```
@@ -1,15 +1,23 @@
 name: "Attachment: Filename Containing Unicode Right-to-Left Override Character"
 description: |
-  Idenitifies attachments that attempt to conceal their true file extension by using right-to-left override characters
+  Recursively idenitifies attachments that attempt to conceal their true file extension by using right-to-left override characters
 references:
   - "https://redcanary.com/blog/right-to-left-override/"
   - "https://sublimecommunity.slack.com/archives/C02N1F7FP7C/p1644411026329629"
 type: "rule"
 authors:
   - twitter: "vector_sec"
 source: |
-  type.inbound 
-  and length(attachments) > 0
-  and any(attachments, iregex_search(.file_name, '\x{202E}', '\x{202D}')) 
+  type.inbound
+  and (
+      any(attachments, iregex_search(.file_name, '\x{202E}', '\x{202D}'))
+      or (
+          any(attachments, .file_extension in~ ("7z","bz2","gz","rar","tar","zip","zipx","iso","img")
+              and any(beta.binexplode(.),
+                  iregex_search(.name, '\x{202E}', '\x{202D}')
+              )
+          )
+      )
+  )
```

you can put these two together. not necessarily for performance, but I tend to think of it more like this:
```suggestion
  and any(attachments,
          iregex_search(.file_name, '\x{202E}', '\x{202D}') or (
              .file_extension in~ ("7z","bz2","gz","rar","tar","zip","zipx","iso",  "img")
              and any(beta.binexplode(.),
                  iregex_search(.name, '\x{202E}', '\x{202D}')
              )
          )
      )
```